File: app/src/main/java/com/garden/taskgarden/ShopItem.java
package com.garden.taskgarden;


/**
 * Class to represent an item in the shop.
 * If you don't supply a resource name to the constructor, will just use the name of the item
 * as a resource name.
 *
 * @author Blake MacDade
 * */
public class ShopItem {

    private String Name;
    private String Description;
    private int Price;
    private String ResourceName;

    public ShopItem(String name, String description, int price, String resourceName){
        this.Name = name;
        this.Description = description;
        this.Price = price;
        this.ResourceName = resourceName;
    }

    public ShopItem(String name, String description, int price){
        this.Name = name;
        this.Description = description;
        this.Price = price;
        this.ResourceName = name;
    }

    public String getName(){
        return this.Name;
    }

    public String getDescription(){
        return this.Description;
    }
    public int getPrice(){
        return this.Price;
    }

    public String getResourceName(){
        return this.ResourceName;
    }

    public void updateName(String name){
        this.Name = name;
    }

    public void updateDescription(String description){
        this.Description = description;
    }

    public void updatePrice(int price){
        this.Price = price;
    }

    public void updateResourceName(String resourceName){
        this.ResourceName = resourceName;
    }

}

File: app/src/main/java/com/garden/taskgarden/ShopItem.kt
package com.garden.taskgarden

/**
 * Class to represent an item in the shop.
 * If you don't supply a resource name to the constructor, will just use the name of the item
 * as a resource name.
 *
 * @author Blake MacDade
 */
class ShopItem {
    var name: String
        private set
    var description: String
        private set
    var price: Int
        private set
    var resourceName: String
        private set

    constructor(name: String, description: String, price: Int, resourceName: String) {
        this.name = name
        this.description = description
        this.price = price
        this.resourceName = resourceName
    }

    constructor(name: String, description: String, price: Int) {
        this.name = name
        this.description = description
        this.price = price
        resourceName = name
    }

    fun updateName(name: String) {
        this.name = name
    }

    fun updateDescription(description: String) {
        this.description = description
    }

    fun updatePrice(price: Int) {
        this.price = price
    }

    fun updateResourceName(resourceName: String) {
        this.resourceName = resourceName
    }
}